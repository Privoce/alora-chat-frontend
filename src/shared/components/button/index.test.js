import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import {
	ButtonComponent
} from 'shared/components';

describe('<ButtonComponent />', () => {
	const onButtonClick = sinon.spy();

	const modelI = {
		text: 'TesteButton',
		type: 'Submit',
		isFetching: true,
		onClick: onButtonClick
	};

	const modelII = {
		text: 'TesteButton',
		type: 'Submit',
		isFetching: false,
		onClick: onButtonClick
	};

	const getWrapper = (model) => {
		return shallow((
			<ButtonComponent {...model} />
		));
	};

	it('renders correctly', () => {
		const wrapper = getWrapper(modelI);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('has a button component', () => {
		const wrapper = getWrapper(modelI);
		expect(wrapper.find('button').length).toBe(1);
	});

	it('has a spinner with isFetching param === true', () => {
		const wrapper = getWrapper(modelI);
		expect(wrapper.find('LoadingComponent').length).toBe(1);
	});

	it('has no spinner with isFetching param === false', () => {
		const wrapper = getWrapper(modelII);
		expect(wrapper.find('LoadingComponent').length).toBe(0);
	});

	it('onclick button is working', () => {
		const wrapper = getWrapper(modelI);
		wrapper.simulate('click', { stopPropagation() { } });
		expect(onButtonClick.called).toBe(true);
	});
});
