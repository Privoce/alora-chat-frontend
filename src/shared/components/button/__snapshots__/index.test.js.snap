// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent /> renders correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
  style={
    Object {
      "margin": undefined,
      "width": undefined,
    }
  }
  type="Submit"
>
  <LoadingComponent
    type="spinner"
  />
</button>
`;
